core/metacling/src/TCling.cxx

```diff
@@ -1720,6 +1690,45 @@ namespace {
 
 }
 
+////////////////////////////////////////////////////////////////////////////////
+///\returns true if the module map was loaded, false on error or if the map was
+///         already loaded.
+bool TCling::RegisterPrebuiltModulePath(const std::string &FullPath,
+                                        const std::string &ModuleMapName /*= "module.modulemap"*/) const
+{
+   assert(llvm::sys::path::is_absolute(FullPath));
+   Preprocessor &PP = fInterpreter->getCI()->getPreprocessor();
+   FileManager &FM = PP.getFileManager();
+   // FIXME: In a ROOT session we can add an include path (through .I /inc/path)
+   // We should look for modulemap files there too.
+   const DirectoryEntry *DE = FM.getDirectory(FullPath);
+   if (DE) {
+      HeaderSearch &HS = PP.getHeaderSearchInfo();
+      HeaderSearchOptions &HSOpts = HS.getHeaderSearchOpts();
+      const auto &ModPaths = HSOpts.PrebuiltModulePaths;
+      bool pathExists = std::find(ModPaths.begin(), ModPaths.end(), FullPath) != ModPaths.end();
+      if (!pathExists)
+         HSOpts.AddPrebuiltModulePath(FullPath);
+      // We cannot use HS.lookupModuleMapFile(DE, /*IsFramework*/ false);
+      // because its internal call to getFile has CacheFailure set to true.
+      // In our case, modulemaps can appear any time due to ACLiC.
+      // Code copied from HS.lookupModuleMapFile.
+      llvm::SmallString<256> ModuleMapFileName(DE->getName());
+      llvm::sys::path::append(ModuleMapFileName, ModuleMapName);
+      const FileEntry *FE = FM.getFile(ModuleMapFileName, /*openFile*/ false,
+                                       /*CacheFailure*/ false);
+
+     // FIXME: Calling IsLoaded is slow! Replace this with the appropriate
+      // call to the clang::ModuleMap class.
+      if (FE && !gCling->IsLoaded(FE->getName().data())) {
+         if (!HS.loadModuleMapFile(FE, /*IsSystem*/ false))
+            return true;
+         Error("TCling::RegisterPrebuiltModulePath", "Could not load modulemap in %s", ModuleMapFileName.c_str());
```

Comment:
This is now `TObject::Error()` which prefixes the location with the class name giving `"TCling::TCling::RegisterPrebuiltModulePath"`

Reply:
Nice, fixed.